lib/basket.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'takeaway'
+
+class Basket
+
+  attr_reader :basket
+
+  def initialize(menu)
+    @basket = []
+    @menu = menu
+  end
+
+  def view
+    "Your basket total is #{basket_total} and it contains:\n#{basket_list}"
+  end
+
+  def add(item, price = @menu.read[item], quantity = get_quantity(item))
+    @basket << hash(item, price, quantity)
+  end
+
+  private
+
+  def hash(item, price, quantity)
+    { :name => item, :price => price, :quantity => quantity }
+  end
+
+  def get_quantity(item)
+    quantity = 1
+    quantity += @basket[get_index(item)][:quantity] unless new_item(item)
+    quantity
+  end
+
+  def get_index(item)
```

Comment:
Your methods are all concise and you are using delegation habitually.

Reply:
yes, lots of nice, short focussed methods 👍